source/renderer/zettlr-attachments.js

```diff
@@ -39,50 +40,64 @@ const FILETYPES_IMG = [
 class ZettlrAttachments {
   constructor (parent) {
     this._renderer = parent
-    this._container = $('<div>').prop('id', 'attachments').css('display', 'none')
-    this._container.html(`<h1>${trans('gui.attachments')} <clr-icon shape="folder" class="is-solid" id="open-dir-external" title="${trans('gui.attachments_open_dir')}"></clr-icon></h1>`)
-    this._fileContainer = $('<div>').prop('id', 'files')
-    this._bibliographyContainer = $('<div>').prop('id', 'bibliography')
-    this._container.append(this._fileContainer)
-    this._container.append($('<h1>').text(trans('gui.citeproc.references_heading')))
-    this._container.append(this._bibliographyContainer)
-    $('body').append(this._container)
     this._open = false
     this._attachments = []
 
     this._act() // Activate both the directory toggle and the link
   }
 
+  get container () {
+    if (!this._container) {
+      this._container = renderTemplate(`<div id="attachments">
+  <h1>${trans('gui.attachments')} <clr-icon shape="folder" class="is-solid" id="open-dir-external" title="${trans('gui.attachments_open_dir')}"></clr-icon></h1>
+  <div id="files">
+    <p>${trans('gui.no_attachments')}</p>
+  </div>
+  <h1>${trans('gui.citeproc.references_heading')}</h1>
+  <div id="bibliography">
+    <p>${trans('gui.citeproc.references_none')}</p>
+  </div>
+</div>`)
+      document.querySelector('body').append(this._container)
+    }
+
+    return document.getElementById('attachments')
+  }
+
+  get fileContainer () {
+    return document.getElementById('files')
+  }
+
+  get bibliographyContainer () {
+    return document.getElementById('bibliography')
+  }
+
   /**
     * Shows/hides the pane.
     */
   toggle () {
     // Toggles the display of the attachment pane
-    if (!this._open) {
-      this._container.css('display', '')
-      this._container.animate({ 'right': '0%' })
-    } else {
-      this._container.animate({ 'right': '-20%' }, () => {
-        this._container.css('display', 'none')
-      })
-    }
-
     this._open = !this._open
+    this.container.classList.toggle('open', this._open)
   }
 
   /**
     * Refreshes the list with new attachments on dir change.
     */
   refresh () {
-    this._fileContainer.text('')
+    if (!this.fileContainer) {
```

Comment:
Please reduce this statement to a one-liner!

Reply:
I'm usually all about shorter code, but I strongly prefer being explicit with blocks for conditions, because they can lead to bugs and security issues. One verify famous one is the `goto fail;` one:

- https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/
- https://www.martinfowler.com/articles/testing-culture.html

Let me know what you think and if you still prefer to have it as a one-liner.

Reply:
While this is none of my business ...

Personally, I think that omitting brackets for one-line blocks (and even removing the newline) _sometimes_ makes the code look cooler (maybe even cleaner), but not all the time (e.g. if the line becomes very long), and there's no other advantage. Well, sure, you get to save two keystrokes. But this "rule" introduces arbitrariness, which I don't like at all.

Always using brackets for blocks and always inserting a newline after a condition is a rule that can be used consistently, and will almost always make the code easier to read. In my humble opinion. :-)

Reply:
> but I strongly prefer being explicit with blocks for conditions, because they can lead to bugs and security issues

Agreed. I personally only tend to do so with these `return`/`continue`-expressions, but as @cdaven said, it's a little bit arbitrary. The thing is that I've used these one-liners all around the code and my inner OCD prevents me from simply leaving it like that. So maybe we can begin to look if we see these one-liners more often and convert them as we encounter them …?

Reply:
I've looked into enforcing brackets for `if` statements for an different codebase in the past, and the ESLint [curly](https://eslint.org/docs/rules/curly) rule with the `"all"` setting is the best way to do it.

Reply:
Okay then let's leave it like this, and enforce that setting at a later time (but please first let us deal with some of the high amount of issues … I want the linter to return green for the tests asap)